showcase/echo_test.go

```diff
@@ -94,6 +99,50 @@ func TestEcho_error(t *testing.T) {
 
 }
 
+// Test dynamic routing header generation. We cannot guarantee the order that headers are sent, so we check that the header sent contains the correct elements as opposed to checking
+// the header itself.
+func TestEchoHeader(t *testing.T) {
+	//defer check(t)
+	for _, tst := range []struct {
+		req  *showcasepb.EchoRequest
+		want []string
+	}{
+		{
+			req:  &showcasepb.EchoRequest{OtherHeader: "projects/123/instances/456"},
+			want: []string{"baz=projects%2F123%2Finstances%2F456", "qux=projects%2F123"},
+		},
+		{
+			req:  &showcasepb.EchoRequest{OtherHeader: "instances/456"},
+			want: []string{"baz=instances%2F456"},
+		},
+		{
+			req:  &showcasepb.EchoRequest{Header: "potato"},
+			want: []string{"header=potato", "routing_id=potato"},
+		},
+		{
+			req:  &showcasepb.EchoRequest{Header: "projects/123/instances/456"},
+			want: []string{"header=projects%2F123%2Finstances%2F456", "routing_id=projects%2F123%2Finstances%2F456", "super_id=projects%2F123", "table_name=projects%2F123%2Finstances%2F456", "instance_id=instances%2F456"},
+		},
+		{
+			req: &showcasepb.EchoRequest{
+				Header:      "regions/123/zones/456",
+				OtherHeader: "projects/123/instances/456",
+			},
+			want: []string{"baz=projects%2F123%2Finstances%2F456", "qux=projects%2F123", "table_name=regions%2F123%2Fzones%2F456", "header=regions%2F123%2Fzones%2F456", "routing_id=regions%2F123%2Fzones%2F456"},
+		},
+	} {
+		mdForHeaders := metadata.New(map[string]string{})
+		echo.Echo(context.Background(), tst.req, gax.WithGRPCOptions(grpc.Header(&mdForHeaders)))
+		got := mdForHeaders.Get("x-goog-request-params")
+		got = strings.Split(got[0], "&")
+		sort.Strings(got)
+		sort.Strings(tst.want)
+		if !reflect.DeepEqual(got, tst.want) {
```

Comment:
Nit: use cmp.Diff (look around the codebase for uses) to compare to values in a test. This can be done in another PR though.